test-tool/test_write10_residuals.c

```diff
@@ -75,11 +75,15 @@ test_write10_residuals(void)
                 return;
         }        
         logging(LOG_VERBOSE, "Verify that the target returned SUCCESS");
-        if (task->status != SCSI_STATUS_GOOD) {
+        ok = task->status == SCSI_STATUS_GOOD ||
```

Comment:
This is wrong. A SCSI target must not report "ILLEGAL REQUEST" in case of a residual.

Reply:
Hi @bvanassche,

It is a good practice to separate a refactoring and a changes for the more convenient review.  All repetitions are removed in the next commit. Even if I introduce a function or a macro, it will be removed by the next commit. However, if you think that it is really better to remove the duplication in this commit, please tell me and I will do it.

Thanks!

Reply:
It is a good practice not to introduce repetitive code. Please introduce a helper function in this commit that returns the result of the following expression:

task->status == SCSI_STATUS_GOOD ||
￼                (task->status == SCSI_STATUS_CHECK_CONDITION &&
￼                 task->sense.key == SCSI_SENSE_ILLEGAL_REQUEST &&
￼                 task->sense.ascq == SCSI_SENSE_ASCQ_INVALID_FIELD_IN_INFORMATION_UNIT

Reply:
done